### ISSUE TITLE
Add Logic functions to Numpy Frontend

### ISSUE THREAD

**Issue by Aarsh2001** [unifyai/ivy#1527]
Add [Logic functions](https://numpy.org/doc/stable/reference/routines.logic.html#) to Numpy frontend:

- [ ] all
- [ ] any
- [ ] isfinite
- [ ] isinf
- [ ] isnan
- [ ] isnat
- [ ] isneginf
- [ ] isposinf
- [ ] iscomplex
- [ ] iscomplexobj
- [ ] isfortran
- [ ] isreal
- [ ] isrealobj
- [ ] isscalar
- [ ] logical_and
- [ ] logical_or
- [ ] logical_not
- [ ] logical_xor
- [ ] allclose
- [ ] isclose
- [ ] array_equal
- [ ] array_equiv
- [ ] greater
- [ ] greater_equal
- [ ] less
- [ ] less_equal
- [ ] equal
- [ ] not_equal

**Comment by Migabaj** [comment]
- [ ] #2819

**Comment by teddywaweru** [comment]
#3070

**Comment by teddywaweru** [comment]
#3070

**Comment by teddywaweru** [comment]
#3070

**Comment by athar-usama** [comment]
- #3082

**Comment by omerfarukballi** [comment]
Hi!

I am opening a new issue about the function in here; which is about numpy functions.

I opened the issue of https://github.com/unifyai/ivy/issues/4428 but I couldnot linked. Would you help me?

**Comment by omerfarukballi** [comment]
isreal function update - #5661

**Comment by ndurumo254** [comment]
#7180

**Comment by ndurumo254** [comment]
- [ ]#7217

**Comment by Mukhriddin19980901** [comment]
#8839

**Comment by Mukhriddin19980901** [comment]
#8840

**Comment by dmegkos** [comment]
#8892

**Comment by shireenchand** [comment]
#10448

**Comment by shireenchand** [comment]
#10535

**Comment by zuruoke** [comment]
#10626

**Comment by Sarvesh-Kesharwani** [comment]
isrealobj #11171

**Comment by Sarvesh-Kesharwani** [comment]
iscomplexobj #12032

**Comment by Sarvesh-Kesharwani** [comment]
isreal #12058

**Comment by shireenchand** [comment]
#12105

**Comment by Sarvesh-Kesharwani** [comment]
isreal

**Comment by Sarvesh-Kesharwani** [comment]
iscomplexobj

**Comment by Sarvesh-Kesharwani** [comment]
iscomplex

**Comment by Lancelot03** [comment]
Hi, we have to make new folder for isreal and isrealobj or we have to do it in truth_value_testing.py

**Comment by Sarvesh-Kesharwani** [comment]
> Hi, we have to make new folder for isreal and isrealobj or we have to do it in truth_value_testing.py

i think should be added in truth_value_testing module
as these functions are closely related to this module than others

and it seems ivy auto assign task bot is not working properly
because i have already raised issues and PRs for these functions already.

so you can work on others subtasks.😀

**Comment by Daniel4078** [comment]
> > Hi, we have to make new folder for isreal and isrealobj or we have to do it in truth_value_testing.py
> 
> i think should be added in truth_value_testing module as these functions are closely related to this module than others
> 
> and it seems ivy auto assign task bot is not working properly because i have already raised issues and PRs for these functions already.
> 
> so you can work on others subtasks.😀

The reason why the bot is not working is because you are not referencing your issues in the correct format! It should be
"- [ ] #issue_number"

**Comment by Sarvesh-Kesharwani** [comment]
- [ ] #12379

**Comment by Sarvesh-Kesharwani** [comment]
> > > Hi, we have to make new folder for isreal and isrealobj or we have to do it in truth_value_testing.py
> > 
> > 
> > i think should be added in truth_value_testing module as these functions are closely related to this module than others
> > and it seems ivy auto assign task bot is not working properly because i have already raised issues and PRs for these functions already.
> > so you can work on others subtasks.😀
> 
> The reason why the bot is not working is because you are not referencing your issues in the correct format! It should be "- [ ] #issue_number" also, please finish working with one issue before trying to claim another one!

actually i reserved next issue only after raising it's pull request for current one. 
thanks for the information though.

**Comment by akstar-007** [comment]
- [ ] #13893

**Comment by akstar-007** [comment]
Hi, I opened the issue to work on the `iscomplex` method. But in the ivy codebase I find the code to be implemented along with its test. What should I do, can anybody please help?

**Comment by utkarsh-shrivastav77** [comment]
- [ ] #22605

**Comment by kingjuno** [comment]
- [ ] #22766

**Comment by AsmaaMHadir** [comment]
- logical_and #22835

**Comment by omartonsy02** [comment]
[logical_xor] #23126

**Comment by omartonsy02** [comment]
> * [logical_xor] [logical_xor #23126](https://github.com/unifyai/ivy/issues/23126)

I found this issue already done.

**Comment by khawajaJunaid** [comment]
- [] #23415

**Comment by Hamzamazhar1999** [comment]
- [ ] #25788

**Comment by adil275** [comment]
- [ ] #25787

**Comment by muhammadhammad-tech** [comment]
#25798

**Comment by keshavkrishna** [comment]
-[]#26021
